core/src/com/ericc/the/game/systems/realtime/RenderSystem.java

```diff
@@ -124,20 +111,31 @@ public void update(float deltaTime) {
 
     private void drawEntity(Entity entity) {
         PositionComponent pos = Mappers.position.get(entity);
-        SpriteSheetComponent render = Mappers.spriteSheet.get(entity);
+        RenderableComponent render = Mappers.renderable.get(entity);
+
+        float lightLevel = render.lightLevel;
+        if (lightLevel == 0.0f) {
+            return;
+        }
 
-        transform.idt();
-        transform.translate(-render.sprite.getOriginX(), -render.sprite.getOriginY()); // From bottom-left-corner space to origin space.
-        transform.mul(render.transform); // Apply affine animations.
-        transform.translate(pos.x, pos.y); // Move to logical position.
+        transformTmp.idt();
+        transformTmp.mul(render.model.defaultTransform); // From bottom-left-corner space to origin space.
+        transformTmp.mul(render.transform); // Apply affine animations.
+        transformTmp.translate(pos.x, pos.y); // Move to logical position.
 
-        batch.draw(render.sprite, render.sprite.getWidth(), render.sprite.getWidth(), transform);
+        float brightness = getBrightness(lightLevel);
+        batch.setColor(colorTmp.set(Color.WHITE).mul(brightness, brightness, brightness, 1.0f));
+        batch.draw(render.region, render.model.width, render.model.height, transformTmp);
     }
 
     private void drawTile(SpriteBatch batch, int x, int y, boolean isStatic) {
 
+        float lightLevel = map.light[x][y];
```

Comment:
Czysto syntaktycznie - lightLevel i brightness to to samo. Warto przemianować na coś w stylu spriteBrightness bo jak rozumiem brightness dotyczy samego sprite'a, a lightLevel otoczenia.